flow/src/main/java/com/sequenceiq/flow/domain/FlowChainLog.java

```diff
@@ -82,6 +89,13 @@ public String getChain() {
         return chain;
     }
 
+    public Queue<Selectable> getChainAsQueue() {
+        if (StringUtils.isEmpty(chain)) {
+            return new ConcurrentLinkedDeque<>();
+        }
+        return (Queue<Selectable>) JsonReader.jsonToJava(chain);
```

Comment:
Please discuss with @bergerdenes about this.

Reply:
Anyway it looks good to me!

Reply:
Please use similar double deserialization as it is present in the `FlowChainHandler.restoreFlowChain`

Reply:
I put that double deserialization into a method in FlowChainLog.